### ISSUE TITLE
How to load the linear part in my code

### ISSUE THREAD

**Issue by fdszy** [vturrisi/solo-learn#181]
Hello! I am new to machine learning and very appreciate your job. I have finished training and I donot know how to load the linear model in another file. Currently I load the Linear with the code below but the accuracy is Inconsistent with training process. 

```python
class linear(nn.Module):
    def __init__(self,n_features,n_classes):
        super(linear,self).__init__()
        self.model = nn.Linear(n_features,n_classes)

    def forward(self,x):
        return self.model(x)

linear_model = linear(model.inplanes,catagory_num).to(device)
linear_state = torch.load("linear.ckpt")["state_dict"]
linear_model.load_state_dict(linear_state,strict=False)
```

**Comment by fdszy** [comment]
I changed my code but it still cannot work

```python
    linear_model = linear(model.inplanes,catagory_num).to(device)
    linear_state = torch.load("linear.ckpt")["state_dict"]
    print(linear_state.keys())
    linear_state = {k:v for k,v in linear_state.items() if "classifier" in k}
    linear_model.load_state_dict(linear_state,strict=False)
    linear_model = linear_model.to(device)
```

**Comment by DonkeyShot21** [comment]
Hi, can you explain a bit better what you are trying to do? Is there a specific reason why you didn't use `main_linear.py`

**Comment by fdszy** [comment]
I have used main_linear.py to train the linear and then I want to load the linear and the encoder in another file. But I when I loaded them(ckpt files) with the above codes, the accuracy drop to only 1% in CIFAR100.

I use another files to load the linear because I need to use the contrastive models in another project.

Thanks a lot for your help.

**Comment by DonkeyShot21** [comment]
I think the easiest way to do that is to pass a dummy backbone:
````
linear = METHODS["linear"](backbone=resnet18())
```
and then load a checkpoint using pytorch lightning:
```
linear.load_from_checkpoint("path_to_checkpoint")
```

**Comment by vturrisi** [comment]
Just to complement, what you are doing is just trying to load the linear classifier, and not the whole model (encoder + classifier). Also, there's a mismatch in names so the classifier is not being loaded as well. If you remove `strict=False` you will see an error poping up.

**Comment by fdszy** [comment]
There is no error when I remove strict=False. But the accuracy is too low. I am sure I am using the correct test data set.

**Comment by fdszy** [comment]
Ok, I have solved the problem, Thanks a lot.
